virtool/users/sessions.py

```diff
@@ -86,34 +88,38 @@ async def get_session(
     Will return `None` if the session doesn't exist or the session id and token do not
     go together.
 
-    :param db: the application database client
+    :param redis: the application redis client
     :param session_id: the session id
     :param session_token: the token for the session
     :return: a session document
 
     """
-    document = await db.sessions.find_one({"_id": session_id})
 
-    if document is None:
+    unparsed_session = None
+    if session_id:
+        unparsed_session = await redis.get(session_id)
+
+    if not unparsed_session:
         return None, None
 
+    session = json.loads(unparsed_session)
```

Comment:
Do you know what will happen if no session matching the id is found?

If `redis.get()` just returns `None` when a value is not found, then this code can be simplified a bit.

If an exception is raised, handling is missing.

Reply:
Nevermind. Simplfication is not possible.

Reply:
It returns `None` if the key is not found